### PR TITLE
feat(tag): added short tag

### PR DESCRIPTION
**Pull request by alembe**
Added a prop for a short tag to fix the height (18px vs 24px before)
Based off of this issue: https://github.com/carbon-design-system/carbon-contribution/issues/23

### CODE REVIEW

**Comment by abbeyhrt** [comment]
Hi @alembe! Thank you for taking the time to contribute!  Based on our [versioning guide](https://github.com/carbon-design-system/carbon/blob/master/docs/guides/versioning.md#a-new-prop-is-added-to-a-component), we would consider this change a feature and not a fix, would you mind changing the PR title to `feat(tag): added short tag`? Thank you! 🙏

**Comment by abbeyhrt** [comment]
@alembe Noticed that the close button outsizes the short Tag on hover, could you address that if you get the chance? 
Screen Shot 2020-01-06 at 11 25 50 AM

**Comment by alembe** [comment]
@abbeyhrt Fixed the height of the tag on hover, let me know if you need anything else from me!

**Comment by alembe** [comment]
@abbeyhrt I fixed the height of the svg on hover, let me know if there's anything else you want me to do!

**Review by aagonzales — COMMENTED** [review]
Looking good, just the hover of the close should be a circle. It's a little oval-y right now. It should also be inset a little so the circle size should be 14x14px.
image

**Comment by alembe** [comment]
@aagonzales I fixed the circle but 14x14 looks a little small with the x in the middle
Screen Shot 2020-01-06 at 2 58 42 PM

**Review by dakahn — APPROVED** [review]
once design signs off we're good! 👍

File: packages/react/src/components/Tag/Tag.js
Comment (asudoh) [inline packages/react/src/components/Tag/Tag.js]: Wondering if we can make `size: PropTypes.oneOf(['regular', 'short'])`. We used to have `small: PropTypes.bool` in `<Button>` but we had to deprecate that prop due to additional size coming up: https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/components/Button/Button.js#L131-L144

**Review by asudoh — COMMENTED** [review]
Thank you for your contribution @alembe!

**Comment by aagonzales** [comment]
Can you make the x larger but keep the circle that smaller size?

**Comment by alembe** [comment]
@aagonzales It doesn't look like it because the x is apart of the circle svg

**Comment by aagonzales** [comment]
Ok then let's make the circle 18x18 and sit flush to the edge of the tag

Image demonstrating only positioning. 
image

**Comment by alembe** [comment]
@aagonzales The filter tag doesn't have that at the end of the tag, there's still space at the end. Do you want that change to filter tag as well?
Screen Shot 2020-01-07 at 10 08 58 AM

**Comment by aagonzales** [comment]
@alembe Yes, it would be great to make it look exactly like the filter tag but the only way to do that is make the circle smaller correct and then we run into the problem of the small x.

**Comment by aagonzales** [comment]
Oh wait you mean change the filter tag to be flush, no we do not want to change that. That's the ideal design.

**Comment by alembe** [comment]
@aagonzales I found the removing the padding around the svg, the x looks more normal
Screen Shot 2020-01-07 at 10 13 48 AM

To be honest, I don't know what you mean by having the filter tag to be flush haha

**Comment by aagonzales** [comment]
Ok sorry this should have all been spec'ed in the original issue but wasn't. 
The circle should be 16x16 which will give it a 1px inset which will make it look like the filterable icon. The focus border should only be 1px.
image

**Comment by alembe** [comment]
@abbeyhrt The 16x16 circle still doesn't have space but a 14x14 does it 
Screen Shot 2020-01-07 at 11 15 03 AM
vs 
Screen Shot 2020-01-07 at 11 13 33 AM

**Comment by aagonzales** [comment]
Oh must be something about how the circle is drawn in sketch vs code. Yeah that looks almost correct but it needs to move right 1px so its equal all the way around.

**Comment by alembe** [comment]
@aagonzales I think this is good to go, let me know if you have any more additions!

**Comment by aagonzales** [comment]
@alembe The react deploy preview failed so I can't review it.

**Review by abbeyhrt — APPROVED** [review]
Looks great! Thank you @alembe!

**Comment by asudoh** [comment]
@alembe A reminder on my review comment: https://github.com/carbon-design-system/carbon/pull/4953/files#r363491401

Please don't hesitate to bring up if you have any questions. Thanks!

**Comment by joshblack** [comment]
@aagonzales do we need to worry about touch target size with this?

**Comment by aagonzales** [comment]
Its most like in a data table which will already have touch target problems. In fact our normal tags probably have touch target issues.

**Comment by joshblack** [comment]
Is there anything we can do to remedy it? I would be worried if we added this and then had to backtrack because of the route we'd have to go down to solve the touch target problem. Unless  addressing the issue isn't that big of a deal? Just curious

**Comment by aagonzales** [comment]
We can set an invisible hit target of 44px then just leave it up to design teams to make sure nothing else in the that target area. It will already be a problem with table rows though which are 32px.

**Comment by joshblack** [comment]
@aagonzales is there a specific article/resource you're thinking of for the invisible target size? If I understand correctly, we can implement something like that with padding but it would impact the overall layout as a result. If the layout doesn't grow, then we would have overlapping targets in the data table use-case.

**Comment by aagonzales** [comment]
Hmm no I was hoping you'd have suggestions. The acceptable touch target is 44px which if we had to make the actual tag that size it would defeat the purpose of this issue.

File: packages/react/src/components/Tag/Tag.js
Comment (asudoh) [inline packages/react/src/components/Tag/Tag.js]:
```suggestion
    [`${prefix}--tag--short`]: size === 'short',
```

File: packages/react/src/components/Tag/Tag.js
Comment (asudoh) [inline packages/react/src/components/Tag/Tag.js]:
```suggestion
  size,
```

File: packages/react/src/components/Tag/Tag.js
Comment (asudoh) [inline packages/react/src/components/Tag/Tag.js]:
```suggestion
  size: PropTypes.oneOf(['regular', 'short']),
```

**Review by asudoh — COMMENTED** [review]
@alembe Feel free to let us know if you no longer have time to make this update. Thanks!

File: packages/react/src/components/Tag/Tag-story.js
Comment (emyarod) [inline packages/react/src/components/Tag/Tag-story.js]: going off of @asudoh's suggested changes in `Tag.js`

```suggestion
      <Tag className="some-class" {...props.filter()} filter size="short">
```

File: packages/components/src/components/tag/_tag.scss
Comment (emyarod) [inline packages/components/src/components/tag/_tag.scss:150]: should we use Carbon spacing tokens for this value?

**Review by emyarod — COMMENTED** [review]
is the short size available for filter tags only or should there also be an example of non-filter short tags in the storybook?

**Comment by asudoh** [comment]
@alembe Are you still pursuing for completing this PR? Thanks!

**Comment by vsnichols** [comment]
@emyarod we need the short size for both filterable and non-filterable tags

**Comment by vsnichols** [comment]
Thanks to everyone for moving this forward and helping guide the implementation! 

@joshblack The short tag is needed for the short data table use case (32px rows). As @aagonzales  noted, the table in general has a 32px target, not 44px. We would have to decide to stop supporting a condensed view to address this, and we know users want the option for a condensed view.

Gmails "compact" view for example looks like it is also about 32px if not denser. 

As long as we are providing users with the option to set table density, which we will be doing with the new data table component settings menu, this should not be an issue.

**Comment by joshblack** [comment]
@vsnichols makes sense! Thanks.

**Comment by aagonzales** [comment]
The react preview deploys are failing and I don't have a local build set up so @laurenmrice can you take a look at this in your local build and approve if it looks good. (sorry for passing the buck).

**Comment by aagonzales** [comment]
@emyarod were all your concerns fixed?

**Review by laurenmrice — CHANGES_REQUESTED** [review]
@alembe The tag height looks like its currently set at 16px, it should be 18px. This may resolve some of the hover issues

File: packages/react/src/components/Tag/Tag.js
Comment (asudoh) [inline packages/react/src/components/Tag/Tag.js:96]:
```suggestion
  size: PropTypes.oneOf(['regular', 'short']),
```

**Comment by vsnichols** [comment]
@alembe I think the last change that needs to be made is to make sure the short tag is 18px high (looks like it is 16px currently). Then hopefully we should be good to go...

**Review by laurenmrice — CHANGES_REQUESTED** [review]
Tag height looks good 👍🏻

The hover circle and focus are nudged too far to the right. The hover circle should be 16px x 16px with a 1px padding between the hover and the tag background.

Spec:
Artboard

**Comment by joshblack** [comment]
Hi @alembe! 👋 Let us know if you're still interested in pursuing this!

**Comment by vsnichols** [comment]
@joshblack we definitely are - @alembe sorry there is one more requested change - see below

**Comment by joshblack** [comment]
Hi there! 👋 We're going through older PRs and closing ones that have gone stale. Let me know if you still want to work on this and we'd be happy to re-open!

**Pull request closed without merging**
